src/optionsdialog.cpp

```diff
@@ -179,6 +179,9 @@ void OptionsDialog::writeSettings(void)
     settings->endGroup();
     ///-----PROGRAMING--LANGUAGE---------------------------------------------------
     settings->beginGroup("compilation_settings");
+    settings->setValue("defaultLanguage/text",defaultLanguageComboBox->currentText());
```

Comment:
Ok, I see your point. You could leave it if you will need it in a future. Generally, please, avoid adding thing if you not sure it will be nedded. There is a danger to leave some unused code, that only make program more complicated than needed. You could always add anything later in a scope of future task.